src/main/scala/edu/ie3/simona/service/weather/WeatherService.scala

```diff
@@ -273,26 +278,39 @@ object WeatherService extends SimonaService {
     // get the weather and send it to the subscribed agents
     // no sanity check needed here as we can assume that we always have weather available
     // when we announce it. Otherwise, the registration would have failed already!
-    updatedStateData.weatherSource
-      .getWeather(tick, updatedStateData.weightedWeatherCoordinates)
-      .foreach { case coordinate -> weatherResult =>
-        updatedStateData.coordsToActorRefMap
-          .get(coordinate)
-          .foreach(recipients =>
-            recipients.foreach(
-              _ ! DataProvision(
-                tick,
-                ctx.self,
-                weatherResult,
-                maybeNextTick,
-              )
+    updatedStateData.coordinateData.foreach { case (_, coordinateData) =>
+      val coordinateWeights = coordinateData.coordinateWeights
+
+      coordinateData.registeredActors.foreach { case (dataType, actors) =>
+        val weatherData = dataType match {
+          case WeatherDataType.Current =>
+            updatedStateData.weatherSource.getWeather(tick, coordinateWeights)
+          case WeatherDataType.CurrentAndForecast(length, interval) =>
```

Comment:
The parameter `interval` is not used. Will this be addressed in another issue or can it be removed?

Reply:
Good point. Implemented a method that uses the interval to reduce the time series resolution.